packages/icon/src/createIconComponent.tsx

```diff
@@ -32,20 +25,15 @@ export default function createIconComponent<
   const Icon = ({ glyph, size = Size.Default, ...rest }: IconProps) => {
     const SVGComponent = glyphs[glyph];
 
-    return (
-      <SVGComponent
-        {...rest}
-        size={typeof size === 'number' ? size : sizeMap[size]}
-      />
-    );
+    return <SVGComponent {...rest} size={size} />;
   };
 
   Icon.displayName = 'Icon';
 
   Icon.propTypes = {
     glyph: PropTypes.oneOf(Object.keys(glyphs)).isRequired,
     size: PropTypes.oneOfType([
-      PropTypes.oneOf(Object.keys(sizeMap)),
+      PropTypes.oneOf(Object.values(Size)),
```

Comment:
This should also accept a number or a string, right?

Reply:
number or specific string value, not any string value